Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py

```diff
@@ -2376,43 +2552,61 @@ def fetch_incidents():
                 for incident in demisto.get(raw_res, "resources"):
                     incident['incident_type'] = incident_type
                     incident_to_context = incident_to_incident_context(incident)
-                    incident_date = incident_to_context['occurred']
-
-                    incident_date_timestamp = int(parse(incident_date).timestamp() * 1000)
-
-                    # Update last run and add incident if the incident is newer than last fetch
-                    if incident_date_timestamp > last_fetch_timestamp:
-                        last_fetch_timestamp = incident_date_timestamp
-                        new_last_incident_fetched = incident.get('incident_id')
-
-                    if last_incident_fetched != incident.get('incident_id'):
-                        incidents.append(incident_to_context)
-
-            if len(incidents) == INCIDENTS_PER_FETCH:
-                current_fetch_info_incidents['offset'] = offset + INCIDENTS_PER_FETCH
-                current_fetch_info_incidents['last_fetched_incident'] = new_last_incident_fetched
-            else:
-                current_fetch_info_incidents['offset'] = 0
-                current_fetch_info_incidents['last_fetched_incident'] = new_last_incident_fetched
+                    incidents.append(incident_to_context)
 
             incidents = filter_incidents_by_duplicates_and_limit(incidents_res=incidents, last_run=current_fetch_info_incidents,
                                                                  fetch_limit=fetch_limit, id_field='name')
             for incident in incidents:
                 occurred = dateparser.parse(incident["occurred"])
                 if occurred:
                     incident["occurred"] = occurred.strftime(DATE_FORMAT)
-
+                    demisto.debug(f"CrowdStrikeFalconMsg: Incident {incident['name']} occurred at {incident['occurred']}")
             last_run = update_last_run_object(last_run=current_fetch_info_incidents, incidents=incidents,
                                               fetch_limit=fetch_limit,
                                               start_fetch_time=start_fetch_time, end_fetch_time=end_fetch_time,
                                               look_back=look_back,
                                               created_time_field='occurred', id_field='name', date_format=DATE_FORMAT)
             current_fetch_info_incidents.update(last_run)
+        demisto.debug(f"CrowdstrikeFalconMsg: Ending fetch incidents. Fetched {len(incidents)}")
 
-    demisto.debug(f"CrowdstrikeFalconMsg: Ending fetch incidents. Fetched {len(incidents)}")
+    if "IDP Detection" in fetch_incidents_or_detections:
```

Comment:
This should be 3 seperated 'if' statements because we can get any combinations of the trio.
Also, fetch_incidents_or_detections is a parameter with pre-defined values.